reactor-netty-http/src/main/java/reactor/netty/http/client/MicrometerHttpClientMetricsHandler.java

```diff
@@ -113,8 +113,8 @@ protected void startWrite(HttpRequest msg, Channel channel, @Nullable ContextVie
 
 		responseTimeHandlerContext = new ResponseTimeHandlerContext(recorder, msg, path, channel.remoteAddress());
 		responseTimeObservation = Observation.createNotStarted(recorder.name() + RESPONSE_TIME, responseTimeHandlerContext, OBSERVATION_REGISTRY);
-		if (contextView != null && contextView.hasKey(ObservationThreadLocalAccessor.KEY)) {
-			responseTimeObservation.parentObservation(contextView.get(ObservationThreadLocalAccessor.KEY));
+		if (contextView != null && contextView.hasKey(OBSERVATION_KEY)) {
```

Comment:
I would suggest writing a test that actually uses context-propagation (which should in effect update the Reactor context with this specific key) and then you would reuse that entry that is stored by the key you've copy pasted yourself. That way if for some reason anything changes in the context propagation repo in a breaking way then your tests should fail

Reply:
Actually you don't even need that because the Key is there in `ObservationThreadLocalAccessor` which is in micrometer-core!

Reply:
`micrometer-observation` has only optional dependency. If I try to build my project without that dependency I'm receiving

```
> Task :reactor-netty-core:compileJava
error: cannot access ThreadLocalAccessor
  class file for io.micrometer.context.ThreadLocalAccessor not found
  Consult the following stack trace for details.
  com.sun.tools.javac.code.Symbol$CompletionFailure: class file for io.micrometer.context.ThreadLocalAccessor not found
1 error
```

Reply:
let's go with the route of adding a test sanity checking the key is same as in `ObservationThreadLocalAccessor` 👍 
I'll do the same in reactor-core-micrometer